stjude/proteinpaint
# Reuse the geneset code in GDC matrix and hierCluster launchers

## Description
In master, the hierCluster does not react to cohort changes if the initial genes list is empty, the app gets stuck in the `no default genes` message.

~~This branch reuses the code from `launchGdcMatrix` that uses `showGenesetEdit()` to prompt the user to specify a non-empty initial geneset. Currently marked as draft because it may require a lot more testing.~~ Marked as ready for review: now using a cleaner implementation to use a reactive geneset component that is a part of plotApp.

To test:
- [OncoMatrix](http://localhost:3000/example.gdc.matrix.html?maxGenes=3&cohort=APOLLO-LUAD): comment out lines ~~223-227 in `launchGdcMatrix.js`~~ 82-85 of `client/plots/geneset.js`, then refresh the page. The geneset edit UI should show up. Submitting a new geneset or changing the cohort should render a valid matrix.
- [Gene Expression](http://localhost:3000/example.gdc.exp.html?&cohort=APOLLO-LUAD): Starting with APOLLO-LUAD, the geneset edit UI should also show up. Changing the cohort should render a valid hierCluster. Submitting a geneset will not work for APOLLO-LUAD in qa-orange, there does not seem to be expression data in that case.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
